### ISSUE TITLE
Shift + Enter / Shift + Tab doesn't;t work on Mac Monterey

### ISSUE THREAD

**Issue by Kingdom-Of-Heaven** [jonsequitur/dotnet-repl#50]
I've just installed botnet reply on my Mac Monterey 12.4. According to documentation SHIFT + Enter should start multiline nevertheless id doesn't;t work. Same with Shift+Tab to see last line from history it also doesn't;t work. Is it a bug or am I missing something?

**Comment by jonsequitur** [comment]
`Shift+Enter` does seem to be  broken on Mac.

`Shift+Tab` doesn't show history, it's intended to navigate through completions. 

If you type `Console.<Tab>` it cycles forward through completions and if you type `Console.<Shift+Tab>` it cycles backwards...

In theory. This seems to be broken on the Mac and on Windows, I'm seeing the order of the completions is wrong.

**Comment by Kingdom-Of-Heaven** [comment]
I agree, this is a bug hopefully dev team will fix it asap as it's annoying to work without that basics shortcuts.

**Comment by DamianReeves** [comment]
Any update on this? On a Mac and using F# and no `Shift-Enter` makes this a pain

**Comment by cartercanedy** [comment]
Currently I'm getting a total crash when I press `Tab` or `Shift-Tab`, `Shift-Enter` works for me though.

**Comment by cartercanedy** [comment]
Maybe this was already determined to be a dotnet interactive api bug, but debugging the dotnet interactive source is showing that the bug is in the command parser within Microsoft.DotNet.Interactive.Kernel. Not fixable here @jonsequitur

**Comment by cartercanedy** [comment]
I made the fix in dotnet-interactive upstream, but the tnterface has changed significantly since the package version referenced in the project file. It's going to take some work to update

**Comment by jonsequitur** [comment]
My understanding is this is a .NET console API bug.

@cartercanedy, can you point me to a fix that you tried in .NET Interactive?

**Comment by cartercanedy** [comment]
See [this commit](https://github.com/dotnet/interactive/commit/55cd7c41d32e7199b3c0fdaf4a5d2332ff58aa1a)

**Comment by jonsequitur** [comment]
Thanks!

**Comment by cartercanedy** [comment]
Just confirming that the process doesn't crash after the most recent release, don't know if the issue above was also a result of the same kind of kernel crashes or smth else
